mbus-api/src/mbus_nats.rs

```diff
@@ -99,8 +112,37 @@ impl Bus for NatsMessageBus {
         &self,
         channel: Channel,
         message: &[u8],
+        options: Option<TimeoutOptions>,
     ) -> io::Result<BusMessage> {
-        self.connection.request(&channel.to_string(), message).await
+        let channel = &channel.to_string();
+
+        let options = options.unwrap_or_else(|| self.timeout_options.clone());
+        let mut timeout = options.timeout;
+        let retries_max = options.max_retries;
+        let mut retries = 0;
+
+        loop {
+            let request = self.connection.request(channel, message);
+
+            let result = tokio::time::timeout(timeout, request).await;
+            if let Ok(r) = result {
+                return r;
+            }
+            if Some(retries) == retries_max {
+                return Err(io::ErrorKind::TimedOut.into());
+            }
+            retries += 1;
+            timeout = std::cmp::min(
+                Duration::from_secs(1) * retries,
+                Duration::from_secs(10),
+            );
+
+            log::trace!(
+                "Retrying message bus message on {} with timeout of {:?}",
```

Comment:
Nitpicking but how about ”timeout on {}....”, channel and add the num of retries left? 

Whatever the case bus message on {channel} sounds odd?

Reply:
yeah, sounds a bit like inception :)